src/instructlab/model/evaluate.py

```diff
@@ -799,25 +806,28 @@ def evaluate(
                 system_prompt=system_prompt,
             )
 
-            server = None
-            try:
-                server, api_base, _ = launch_server(
-                    eval_serve=ctx.obj.config.serve,
-                    tls_client_cert=ctx.params["tls_client_cert"],
-                    tls_client_key=ctx.params["tls_client_key"],
-                    tls_client_passwd=ctx.params["tls_client_passwd"],
-                    tls_insecure=ctx.params["tls_insecure"],
-                    model=model,
-                    model_name=model,
-                    max_workers=None,
-                    gpus=gpus,
-                    backend=backend,
-                    enable_serving_output=enable_serving_output,
-                )
-                overall_score, individual_scores = evaluator.run(api_base)
-            finally:
-                if server is not None:
-                    server.shutdown()
+            if skip_server:
```

Comment:
Since the complexity of this portion is growing but we're duplicating the same block in multiple places, we should extract this into a separate function to make it more readable/maintainable.

Reply:
No you don't need a separate PR for this. You can just extract your the entirety of this try/catch structure that you're changing which is common to both portions and place it into a single function.